Mathlib/RingTheory/Coalgebra/GroupLike.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2025 Yaël Dillies, Michał Mrugała. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Michał Mrugała
+-/
+import Mathlib.RingTheory.Coalgebra.Equiv
+import Mathlib.RingTheory.Flat.Domain
+
+/-!
+# Group-like elements in a coalgebra
+
+This file defines group-like elements in a coalgebra, ie elements `a` such that `η a = 1` and
+`Δ a = a ⊗ₜ a`.
+
+## Main declarations
+
+* `IsGroupLikeElem`: Predicate for an element in a coalgebra to be group-like.
+* `linearIndepOn_isGroupLikeElem`: Group-like elements over a domain are linearly independent.
+-/
+
+open Coalgebra Function TensorProduct
+
+variable {F R A B : Type*}
+
+section CommSemiring
+variable [CommSemiring R] [AddCommMonoid A] [AddCommMonoid B] [Module R A] [Coalgebra R A]
+  [Module R B] [Coalgebra R B] {a b : A}
+
+variable (R) in
+/-- A group-like element in a coalgebra is an element `a` such that $\eta a = 1$ and
```

Comment:
```suggestion
/-- A group-like element in a coalgebra is an element `a` such that $\eta(a) = 1$ and
```
is presumably how this would be written on paper?